cli/tests/integration/cache_tests.rs

```diff
@@ -107,3 +108,57 @@ itest!(package_json_basic {
   copy_temp_dir: Some("package_json/basic"),
   exit_code: 0,
 });
+
+// Regression test for https://github.com/denoland/deno/issues/17299
+#[tokio::test]
+async fn cache_put_overwrite() {
+  let test_context = TestContextBuilder::new().use_temp_cwd().build();
+  let temp_dir = test_context.temp_dir();
+
+  let part_one = r#"
+  const req = new Request('http://localhost/abc');
+  const res1 = new Response('res1');
+  const res2 = new Response('res2');
+
+  const cache = await caches.open('test');
+
+  await cache.put(req, res1);
+  await cache.put(req, res2);
+
+  const res = await cache.match(req).then((res) => res?.text());
+  console.log(res);
+    "#;
+
+  let part_two = r#"
+  const req = new Request("http://localhost/abc");
+  const res1 = new Response("res1");
+  const res2 = new Response("res2");
+
+  const cache = await caches.open("test");
+
+  // Swap the order of put() calls.
+  await cache.put(req, res2);
+  await cache.put(req, res1);
+
+  const res = await cache.match(req).then((res) => res?.text());
+  console.log(res);
+      "#;
+
+  temp_dir.write("cache_put.js", part_one);
+
+  let run_command =
+    test_context.new_command().args_vec(["run", "cache_put.js"]);
+
+  let output = run_command.run();
+  output.assert_matches_text("res2\n");
+  output.assert_exit_code(0);
+
+  // The wait will surface the bug as we check last written time
+  // when we overwrite a response.
+  tokio::time::sleep(std::time::Duration::from_secs(1)).await;
```

Comment:
nit: you can use `std::time:sleep` here instead and avoid making this test async.